ompi/mca/coll/basic/coll_basic_neighbor_allgather.c

```diff
@@ -63,6 +63,7 @@ mca_coll_basic_neighbor_allgather_cart(const void *sbuf, int scount,
         }
 
         if (MPI_PROC_NULL != srank) {
+            nreqs += 2;
```

Comment:
@bosilca Replace nreqs+=2 by nreqs++ before each request. That way, we'll be able to free exactly the right number of requests in case of failure.